modules/renter/repair.go

```diff
@@ -0,0 +1,142 @@
+package renter
+
+import (
+	"io"
+	"os"
+	"sync"
+	"sync/atomic"
+)
+
+// repair attempts to repair a file by uploading missing pieces to more hosts.
+func (f *file) repair(r io.ReaderAt, hosts []uploader) error {
+	// determine which chunks need to be repaired
+	// TODO: inefficient -- O(2n) on number of pieces
+	present := make([][]bool, f.numChunks())
+	for i := range present {
+		present[i] = make([]bool, f.erasureCode.NumPieces())
+	}
+	for _, fc := range f.contracts {
+		// TODO: ping host, and don't mark pieces if unresponsive?
```

Comment:
that seems sensible, since the renter is the only package using the hostdb anyway. Would make mocking easier as well. My only concern would be bloating of the renter package.